demo.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python3
+
+"""
+py-orca demonstration script
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This script is intended to demonstrate how you can use py-orca to
+process a dataset (in this case, RNA-seq) using a series of workflow
+runs: nf-synstage, nf-core/rnaseq, and nf-synindex.
+
+This script assumes that the following environment variables are set.
+Refer to `.env.example` for the format, including examples.
+    - NEXTFLOWTOWER_CONNECTION_URI
+    - SYNAPSE_CONNECTION_URI
+    - AWS_PROFILE (or another source of AWS credentials)
+"""
+
+import asyncio
+from dataclasses import dataclass
+from pathlib import PurePosixPath
+from textwrap import dedent
+
+import s3fs
+import yaml
+from metaflow import FlowSpec, Parameter, step
+
+from orca.services.nextflowtower import LaunchInfo, NextflowTowerOps
+from orca.services.synapse import SynapseOps
+
+TMPDIR_PREFIX = "s3://orca-service-test-project-tower-scratch/30days"
+OUTDIR_PREFIX = "s3://orca-service-test-project-tower-bucket/outputs"
+
+
+@dataclass
+class RnaseqDataset:
+    """RNA-seq dataset and relevant details."""
+
+    id: str
+    samplesheet: str
+    output_folder: str
+
+    def get_run_name(self, suffix: str) -> str:
+        """Generate run name with given suffix."""
+        return f"{self.id}_{suffix}"
+
+    def get_staged_samplesheet(self) -> str:
+        """Generate staged samplesheet based on synstage behavior."""
+        path = PurePosixPath(self.samplesheet)
```

Comment:
After reading through the code a couple times, I understand now, but it's a bit unintuitive that dataset.samplesheet changes type. Meaning you pass in synapse id, and it changes to a file path later.

Would it be helpful to throw an error here in case someone doesn't run synstage?

Reply:
Yeah, I'm also unhappy with this part. Ideally, we use `nf-prov` to pull the output files from each workflow run and use that to populate the params for the next step. Here, I'm predicting the location of the staged samplesheet using hard-coded logic. In the very least, this makes more sense in the Flow rather than in the Dataset class because it's "glue logic" that arises from the workflows rather than an intrinsic feature of an RNA-seq dataset.